pkgs/by-name/ut/ut1999/package.nix

```diff
@@ -0,0 +1,122 @@
+{ lib, stdenv, requireFile, autoPatchelfHook, fetchurl,
+libgcc, wxGTK32, innoextract, makeWrapper,
+libGL, SDL2, openal, libmpg123, libxmp }:
+
+let
+  oldUnrealPatch = stdenv.mkDerivation {
+    name = "ut1999-oldunreal-patch";
+    version = "469d";
+    sourceRoot = ".";
+    src = fetchurl {
+      url = "https://github.com/OldUnreal/UnrealTournamentPatches/releases/download/v469d/OldUnreal-UTPatch469d-Linux-amd64.tar.bz2";
+      hash = "sha256-aoGzWuakwN/OL4+xUq8WEpd2c1rrNN/DkffI2vDVGjs=";
+    };
+
+    buildInputs = [
+      libgcc
+      stdenv.cc.cc
+      wxGTK32
+    ];
+
+    nativeBuildInputs = [
+      autoPatchelfHook
+      makeWrapper
+    ];
+
```

Comment:
What are these needed for? From what I can tell, they're not used and this whole thing could also just be a `runCommand`.

Reply:
@Atemu ,

This sub-derivation contains the `OldUnreal` patch which contains a bunch of native executables that need to be patched. The `autoPatchelfHook` applies most of the changes necessary, which require the three `buildInputs`.

```
setting RPATH to: /nix/store/2biv1588jnsdapx2ad8cb0hzyxr172y4-ut1999-oldunreal-patch/System64
auto-patchelf: 7 dependencies could not be satisfied
error: auto-patchelf could not satisfy dependency libwx_gtk3u_core-3.2.so.0 wanted by /nix/store/2biv1588jnsdapx2ad8cb0hzyxr172y4-ut1999-oldunreal-patch/System64/wx-ut-bin-amd64
error: auto-patchelf could not satisfy dependency libwx_baseu-3.2.so.0 wanted by /nix/store/2biv1588jnsdapx2ad8cb0hzyxr172y4-ut1999-oldunreal-patch/System64/wx-ut-bin-amd64
error: auto-patchelf could not satisfy dependency libstdc++.so.6 wanted by /nix/store/2biv1588jnsdapx2ad8cb0hzyxr172y4-ut1999-oldunreal-patch/System64/libfmod.so.13
error: auto-patchelf could not satisfy dependency libgcc_s.so.1 wanted by /nix/store/2biv1588jnsdapx2ad8cb0hzyxr172y4-ut1999-oldunreal-patch/System64/libfmod.so.13
error: auto-patchelf could not satisfy dependency libatomic.so.1 wanted by /nix/store/2biv1588jnsdapx2ad8cb0hzyxr172y4-ut1999-oldunreal-patch/System64/libopenal.so.1
error: auto-patchelf could not satisfy dependency libstdc++.so.6 wanted by /nix/store/2biv1588jnsdapx2ad8cb0hzyxr172y4-ut1999-oldunreal-patch/System64/libopenal.so.1
error: auto-patchelf could not satisfy dependency libgcc_s.so.1 wanted by /nix/store/2biv1588jnsdapx2ad8cb0hzyxr172y4-ut1999-oldunreal-patch/System64/libopenal.so.1
```